data/src/test/java/org/oppia/data/persistence/PersistentCacheStoreTest.kt

```diff
@@ -78,32 +80,43 @@ class PersistentCacheStoreTest {
   @Captor
   lateinit var userAppHistoryResultCaptor2: ArgumentCaptor<AsyncResult<TestMessage>>
 
+  private val backgroundDispatcherScope by lazy {
+    CoroutineScope(backgroundDispatcher)
+  }
+
   @Before
   @ExperimentalCoroutinesApi
   fun setUp() {
     setUpTestApplicationComponent()
-    // The test dispatcher must be paused by default to ensure the blocking executor used by InMemoryCache works
-    // correctly. The dispatcher's default state is to synchronously execute everything that's schedule, but this is not
-    // an accurate simulation of production and results in strange bugs (like impossible, immediate in-place thread
-    // hops that break thread separation assumptions).
-    testDispatcher.pauseDispatcher()
   }
 
   // TODO(#59): Create a test-only proto for this test rather than needing to reuse a production-facing proto.
   @Test
   @ExperimentalCoroutinesApi
```

Comment:
Correct me if I am wrong but, the InternalCoroutinesApi is meant to replace ExperimentalCoroutinesApi right? If so, why are we adding both annotations in the tests? If its not meant to replace ExperimentalCoroutinesApi, why are we adding this annotation as well?

Reply:
The two annotations are unrelated from what I understand. To avoid warnings when building the code, we need ExperimentalCoroutinesApi since downstream the TestCoroutineDispatcher has to have the annotation on its override methods from DelayController.